### PR TITLE
Add typescript definition

### PR DESCRIPTION
**Pull request by skoruba**
This PR added the typescript definition which was descibed in this issue #5.

### CODE REVIEW

**Comment by mifi** [comment]
Hi!
Have you confirmed that microbundle will include the typedefinitions in the dist directory after building? Did you mange to solve the issue that his guy had in #9 ?

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
   "main": "dist/index.js",
   "module": "dist/index.modern.js",
   "source": "src/index.js",
+  "types": "src/types/index.d.ts",
```
Comment (jamiehaywood) [inline package.json:11]: you'll probably need to add this for it to work in the bundling

```suggestion
  "types": "dist/index.d.ts",
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "build": "microbundle-crl --no-compress --format modern,cjs && cp src/types/index.d.ts /dist",
    "start": "microbundle-crl watch --no-compress --format modern,cjs",
```

**Review by jamiehaywood — CHANGES_REQUESTED** [review]
you'll need to add to the end of the `build` script:
`&& cp src/types/index.d.ts dist` (I've not tested this, so check it works)

you'll also need to change the types location to `dist/index.d.ts`

All of this depends on the @mifi 's choice of folder/file structure.

**Comment by mifi** [comment]
that sounds good to me. maybe we should merge the types from this PR with the types from https://github.com/mifi/react-lottie-player/pull/9/files

I see it has some potentially better types like `direction?: 1 | -1` and `animationData: object`

**Comment by jamiehaywood** [comment]
I think we merge them & then re-review, because there are a few types that could be improved

**Comment by osmarpb97** [comment]
Any update?

**Comment by mifi** [comment]
I don't have the TS expertise, but if someone wants to test, verify and vouch for a TS PR, then I'm happy to merge!

**Comment by jamiehaywood** [comment]
@mifi I'm happy to get involved here and raise a PR cherry picking the best types from all 3 PRs.

**Comment by mifi** [comment]
closing in favor of #32
if anyone has improvements to that, feel free to create a pr

**Pull request closed without merging**